docs/dev_guide/compare-prog-models.rst

```diff
@@ -0,0 +1,767 @@
+larger storage space but slower access compared to shared memory. Constant memory
+is a read-only space, suitable for storing unchanging data such as constants or
+lookup tables. Unified memory can be accessed both on host and device.
+
+Similarly, in SYCL, local memory is shared within a work-group, global memory is
+accessible by all work-items, and shared memory is accessible on the host and
+device. According to the `SYCL 2020 Specification <https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html>`_, constant memory no longer appears in the SYCL device memory
+model in SYCL 2020.
+
+Shared Memory
+*************
+
+CUDA shared memory can be mapped to SYCL local memory. To perform this migration,
+declare an accessor with access target set to ``sycl::access::target::local``.
+For example:
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Original CUDA Code
+     - Migrated SYCL Code
+   * - .. code-block::
+
+          __global__ void foo() {
+            __shared__ int shm[16];
+            shm[0] = 2;
+          }
+
+          int main() {
+            foo<<<1, 1>>>();
+          }
+
+          void foo(int *shm) {
+           shm[0] = 2;
+          }
+
+     - .. code-block::
+
+          int main() {
+            sycl::queue q;
+            q.submit([&](sycl::handler &cgh) {
+
+              sycl::local_accessor<int> shm_acc(sycl::range<1>(16), cgh);
+              cgh.parallel_for(
+                  sycl::nd_range<3>(sycl::range<3>(1, 1, 1), sycl::range<3>(1, 1, 1)), [=](sycl::nd_item<3> item_ct1) {
+                    foo(shm_acc.get_pointer());
+                  });
+            });
+          }
+
+Global, Constant, and Unified Memory
+************************************
+
+CUDA global memory and constant memory can be mapped to SYCL global memory. CUDA
+unified memory can be mapped to SYCL shared memory. To perform this migration,
+allocate memory through ``sycl::malloc_device`` or ``sycl::malloc_shared``. For
+example:
+
+.. list-table::
+   :widths: 50 50
+   :header-rows: 1
+
+   * - Original CUDA Code
+     - Migrated SYCL Code
+   * - .. code-block::
+
+          void foo() {
+            int *mem1, *mem2;
+
+
+            cudaMalloc(&mem1, 10);
+            cudaMallocManaged(*mem2, 10);
+          }
+     - .. code-block::
+
+          void foo() {
+            sycl::queue q;
+            int *mem1, *mem2;
+
+            mem1 = sycl::malloc_device<int>(10, q);
+            mem2 = sycl::malloc_shared<int>(10, q);
+          }
+
+Note that in CUDA, type specifiers ``__device__``, ``__constant__``, and ``__managed__``
+can be used to declare a variable resident in global memory and unified memory.
+There is no direct equivalence in SYCL, but you can implement similar functionality
+with helper classes. A reference implementation for ``dpct::global_memory``,
+``dpct::constant_memory``, and ``dpct::shared_memory`` is provided in the
+`SYCLomatic project <https://github.com/oneapi-src/SYCLomatic>`_.

```

Comment:
the link should be https://github.com/oneapi-src/SYCLomatic/blob/4893685d6f2f1c6126a1962be65dfa7688bb4162/clang/runtime/dpct-rt/include/memory.hpp.inc#L1953

Reply:
Done.